Replaced c++ thread pool implementation with a C based one

jorbakk opened the pull request:
The current thread pool implementation used by the software backend is the only C++ code in nanovgXC. Replacing it with the [C based one](https://github.com/Pithikos/C-Thread-Pool/tree/master) from Johan Hanssen Seferidis increases portability and simplifies the build.

jorbakk committed:
Fixed alpha blending images with the software backend

jorbakk committed:
Replaced thread pool implementation with a C based one

pbsurf commented:
The C++ thread pool is fully cross-platform since it just uses the standard library.  I would be fine with a plain C alternative, but it would have to coexist with the C++ one.

jorbakk committed:
Merged changes from master

jorbakk committed:
Introduced single-header library for threadpool

The library uses std-c++ threadpool if compiled as c++, otherwise
if build with a C compiler it uses posix threads.

jorbakk commented:
So here's a proposal using a single-header library for both thread pool implementations. This should build successfully on plain C posix 1995 platforms w/o a C++ compiler and on all platforms with a C++11 compiler. The implementation is chosen by using one of these compilers for building the software backend.

jorbakk committed:
Cosmetics; optionally provide user implementation of threadpool
